compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -1677,16 +1659,14 @@ impl<'a, 'tcx> euv::Delegate<'tcx> for InferBorrowKind<'a, 'tcx> {
         );
 
         // We only want repr packed restriction to be applied to reading references into a packed
-        // struct, and not when the data is being moved. There for we call this method here instead
+        // struct, and not when the data is being moved. Therefore we call this method here instead
         // of in `restrict_capture_precision`.
         let place = restrict_repr_packed_field_ref_capture(
```

Comment:
I'm a *bit* surprised to see this here -- isn't this the only case of "restrictions" and the like that occur in the `borrow` etc methods? I guess we need that because we are already doing a bit of coallescing (i.e., for references to the *exact same place*?). It seems ok, but I have to review the "reference algorithm" to decide if it is indeed equivalent.

Reply:
This is what we talked about here: https://hackmd.io/D3I_gwvuT-SPnJ22tgJumw#Packed-field-ref-and-move

I think we want to maybe re write the inital part of the algoirthm a bit more.